applications/HDF5Application/custom_io/hdf5_connectivities_data.cpp

```diff
@@ -119,7 +122,10 @@ void ConnectivitiesData::CreateEntities(NodesContainerType& rNodes,
         }
         Condition::Pointer p_cond =
             r_cond.Create(mIds[i], nodes, rProperties(mPropertiesIds[i]));
-        rConditions.push_back(p_cond);
+        // here we can safely use the insert with the rConditions.end() as the hint
+        // because, when reading, we can assume that it was written in the
+        // sorted order within HDF5.
+        rConditions.insert(rConditions.end(), p_cond);
```

Comment:
what about out of order creation?

Reply:
Same comment as before

Reply:
same as above, mIds is ordered? but if so, why do you need o create a condition here? are u populating a modelpart  later with this?